src/Runner.php

```diff
@@ -777,7 +777,8 @@ private function processChildProcs($childProcs)
 
         while (count($childProcs) > 0) {
             $pid = pcntl_waitpid(0, $status);
-            if ($pid <= 0) {
+            if ($pid <= 0 || array_key_exists($pid, $childProcs) === false) {
```

Comment:
Just checking - is there any particular reason to use `array_key_exists()` rather than `isset()` for this change ?

Reply:
This mostly boils down to habit and preference.

I'm used to working with strictly typed arrays checked by PHPStan so the values in our projects tend to be guaranteed non-nullable; given that I prefer the clearer intention of array_key_exists but your concern is likely over the additional null check isset() performs.

I can either update the PHPDoc to reflect the contents (however the PHPStan level the project uses doesn't really give any guarantees) or I can change it to isset() instead, which would you prefer?

Reply:
> your concern is likely over the additional null check isset() performs.

Actually, my concern is more one of performance. `isset()` is a language construct, while `array_key_exists()` a function call (and only opcode cached since PHP 7.4).
And yes, while I realize it's a micro-optimization, those type of optimizations actually do have a place in CLI tooling ;-)

> I can either update the PHPDoc to reflect the contents or I can change it to isset() instead, which would you prefer?

If you don't mind, I think I prefer both ;-)